erts/preloaded/src/erlang.erl

```diff
@@ -86,6 +86,7 @@
 -type boolean() :: true | false.
 -type byte() :: 0..255.
 -type char() :: 0..16#10FFFF.
+-type dynamic() :: dynamic().
```

Comment:
Is there any reason why `dynamic()` is not defined as another type alias of `any()`, in the same fashion as `term()`?

**Reasoning**

- Assume that `dynamic()` is not a type alias for `term()` or `any()
- Assume that the standard library replaces `term()` by `dynamic()`, which I think it is not unreasonable (under eqWAlizer and Dialyzer all works)
- Assume there is another static analysis tool. Since `dynamic()` is not a subtype of `any()`, we are breaking backwards compatibility.

Is my reasoning sound?

Reply:
This other tool could be (e.g.,) Gradualizer (https://github.com/josefs/Gradualizer), Etylizer (AFAIK https://arxiv.org/abs/2302.12783), or maybe even the bidirectional typing for Erlang (https://dl.acm.org/doi/10.1145/3471871.3472966)

Reply:
`erlang:dynamic()` is an alias to the builtin type `dynamic()`. Similar to how `erlang:atom()` is an alias to the builtin type `atom()`.

The introduction of the builtin type `dynamic()` is backward-compatible in OTP 26 - thanks to https://github.com/erlang/otp/pull/6335.

Reply:
Note that line 89 doesn't define `dynamic()`, - it "documents" it - https://github.com/erlang/otp/pull/5292.

In light of https://github.com/erlang/otp/pull/6335 - `erlang:dynamic()` and `dynamic()` should be the same.